### PR TITLE
expression: add new scalar function IsTruthWithNull (#19621)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #19621 to release-4.0

---

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/17720, close https://github.com/pingcap/tidb/issues/17476 

Problem Summary: NewFunction can't get keepnull attributes for isture function.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed: 
1. Add a new scalar function name IsTruthWithNull in parser.
2. Use the function name to build ScalarFuncton in wrapWithIsTrue.

How it Works:

### Related changes

- Need to cherry-pick to the release branch 3.0,4.0

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- Fix the behavior of rewrite ScalarFunction IsTure .

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -32,7 +32,11 @@ require (
 	github.com/pingcap/goleveldb v0.0.0-20191226122134-f82aafb29989
 	github.com/pingcap/kvproto v0.0.0-20200818080353-7aaed8998596
 	github.com/pingcap/log v0.0.0-20200828042413-fce0951f1463
+<<<<<<< HEAD
```
Comment (qw4990) [inline go.mod:35]: @wshwsh12 Please fix CI

**Review by lzmhhh123 — COMMENTED** [review]
LGTM

**Comment by XuHuaiyu** [comment]
LGTM

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**